src/rez/package_test.py

```diff
@@ -718,3 +841,163 @@ def print_summary(self):
 
         strs = columnise(rows)
         print('\n'.join(strs))
+
+
+def _is_conflict(*requires):
+    """Check if any of the requirements have a conflict.
+
+    Example:
+        >>> _is_conflict([PackageRequest("foo-2")], [PackageRequest("!foo-2")])
+        # True
+        >>> _is_conflict([PackageRequest("foo-2")], [PackageRequest("foo-1+")])
+        # False
+
+    Args:
+        requires (tuple[iter[PackageRequest]]):
+            A series of Rez packages requests to check for conflicts.
+
+    Returns:
+        bool: If conflict, return True, otherwise, return False.
+
+    """
+    requirement = RequirementList(itertools.chain(*requires))
+
+    return requirement.conflict
+
+
+def _choose_test_object(package, test_requires, extra_packages=frozenset()):
+    """Find the most appropriate Rez object for ``rez-test --interactive``.
+
+    If ``package`` has variants, find a valid variant, given ``test_requires``
+    and ``extra_packages``.
+
+    Args:
+        package (Package):
+            The installed Rez package to query variant / test data from.
+        test_requires (set[PackageRequest], optional)
+            Include these Rez package versions in the check, if provided. If
+            the Rez test has no defined ``"requires"``, this parameter is empty.
+        extra_packages (set[PackageRequest], optional):
+            Include these Rez package versions in the check, if provided.  This
+            is usually empty unless the user explicitly asks for them, e.g.
+            using ``rez-test foo --interactive bar --extra-packages thing-1``.
+
+    Raises:
+        PackageTestError:
+            If ``test_requires`` or ``extra_packages`` result in a Rez request conflict.
+
+    Returns:
+        Package or Variant:
+            Get a specific variant to return. If ``package`` has no variants
+            defined then this function returns ``package``, instead.
+
+    """
+    variants = package.variants
+
+    if not variants:
+        return package
+
+    candidates = []
+
+    for variant_index, package_requests in enumerate(variants):
+        if not _is_conflict(package_requests, test_requires, extra_packages):
+            candidates.append(package.get_variant(variant_index))
+
+    if not candidates:
+        if not extra_packages:
+            raise PackageTestError(
+                "No valid variant could match your given Rez test requires."
+            )
+
+        raise PackageTestError(
+            "No valid variant could match your given Rez test requires + extra packages."
+        )
+
+    return candidates[0]
```

Comment:
At this point in the code, there may be 1-or-more valid variants which can be used. I'm not sure how to "choose" the "best" one so for now I'm just returning the 0th index, the first valid variant. If there's a smarter way to do this, please let me know!